clone.go

```diff
@@ -0,0 +1,192 @@
+package poly
+
+import (
+	"errors"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+)
+
+type CloneSequence struct {
+	Sequence string
+	Circular bool
+}
+
+type Overhang struct {
+	Length   int
+	Position int
+	Forward  bool
+}
+
+type Fragment struct {
+	Sequence        string
+	ForwardOverhang string
+	ReverseOverhang string
+}
+
+type Enzyme struct {
+	EnzymeName        string
+	RegexpFor         *regexp.Regexp
+	RegexpRev         *regexp.Regexp
+	EnzymeSkip        int
+	EnzymeOverhangLen int
+	Directional       bool
+}
+
+/******************************************************************************
+
+Base cloning functions begin here.
+
+******************************************************************************/
+
+// https://qvault.io/2019/10/21/golang-constant-maps-slices/
+func getBaseRestrictionEnzymes() map[string]Enzyme {
+	// Eventually, we want to get the data for this map from ftp://ftp.neb.com/pub/rebase
```

Comment:
What's stopping us from doing this now?

Also, @patrickjyoung is working on an enzyme listing function if that would be helpful here.

Reply:
I also believe that @aleyshachen is working on a sequence to regexp function that would be helpful too.

Reply:
You'll have to write a custom parser for one of their data dumps + you'll need to get sequence (sequence that NEB would provide) -> regexp function.

Reply:
We're gonna have to fix this later.